tests/tests.rs

```diff
@@ -1110,3 +1113,65 @@ fn test_size() {
     // Files with size equal 4 kibibytes.
     te.assert_output(&["", "--size", "+4ki", "--size", "-4ki"], "4_kibibytes.foo");
 }
+
+#[cfg(test)]
+fn create_file_with_modified<P: AsRef<Path>>(path: P, duration_in_secs: u64) {
+    let st = SystemTime::now() - Duration::from_secs(duration_in_secs);
+    let ft = filetime::FileTime::from_system_time(st);
+    fs::File::create(&path).expect("creation failed");
+    filetime::set_file_times(&path, ft, ft).expect("time modification failed");
+}
+
+#[test]
+fn test_modified_relative() {
+    let te = TestEnv::new(&[], &[]);
+    create_file_with_modified(te.test_root().join("0_now"), 0);
+    create_file_with_modified(te.test_root().join("1_min"), 60);
+    create_file_with_modified(te.test_root().join("10_min"), 600);
+    create_file_with_modified(te.test_root().join("1_h"), 60 * 60);
+    create_file_with_modified(te.test_root().join("2_h"), 2 * 60 * 60);
+    create_file_with_modified(te.test_root().join("1_day"), 24 * 60 * 60);
+
+    te.assert_output(
+        &["", "--changed-within", "15min"],
+        "0_now
+        1_min
+        10_min",
+    );
+
+    te.assert_output(
+        &["", "--changed-before", "15min"],
+        "1_h
+        2_h
+        1_day",
+    );
+
+    te.assert_output(
+        &["min", "--changed-within", "12h"],
```

Comment:
This test really confused me for a while because I didn't see the first `"min"` argument. Maybe you could name some of the files `foo_1_h` and others `bar_10_min` and then additionally search for `foo` here?